### ISSUE TITLE
Automatically close a Dropdown when user clicks outside

### ISSUE THREAD

**Issue by dgdavid** [openSUSE/agama#552]
After #545 be merged in master branch, the page options will be shown by using a PatternFly [`Dropdown`](https://www.patternfly.org/v4/components/dropdown) placed close to the hamburger menu icon to open/close the sidebar.

The user can click on the _arrow down_ icon to check open the _menu_ holding the options related to the current _page_. But this dropdown will be closed only when 

* the user closes it by clicking in the arrow again
* the use clicks on any of the available options

Ideally, it should be closed automatically when clicking outside the dropdown itself too (right after losing the focus). At this time there is no prop (or I overlooked it) to tell the component to do so. But, looking at the source code it looks to me that current `PatternFly Dropdown` is/will be deprecated in favor of another that has a `onOpenChange` prop that _might_ help to achieve the mentioned behavior. Needs to check/test it better.

https://github.com/patternfly/patternfly-react/commit/d43cf2427dc9ea1ddb2381284f05276aa2c7cdb7

In any case, we should evaluate a way to automatically close that menu, although it is more a _nice to have_ than a requirement for me.

**Comment by Anandpandey0** [comment]
I can do this . Please assign it to me.

**Comment by dgdavid** [comment]
> I can do this . 

That would be great, **thanks a lot**! Just in case it can help, we were thinking of having a look at the availability of the new PatternFly Dropdown component first.

> Please assign it to me.

Done!

**Comment by dgdavid** [comment]
Please, bear in mind that you have to work on top of the `issues` branch at this moment or... just wait a bit until @joseivanlopez merge it to master.

**Comment by joseivanlopez** [comment]
Yes, you can create a PR through `issues` feature branch. Thanks!

**Comment by dgdavid** [comment]
@Anandpandey0 just out of curiosity, are you finally playing with this one?

**Comment by balsa-asanovic** [comment]
Hey there,

Can I take this over if @Anandpandey0 is no longer interested?

**Comment by imobachgs** [comment]
To be honest, I do not know whether @Anandpandey0 is still interested. I would wait 1 or 2 weeks more and, if there is no response, I would reassign the issue to you, @balsa-asanovic.

**Comment by balsa-asanovic** [comment]
Sure, let's do that

**Comment by imobachgs** [comment]
OK, as @Anandpandey0 is not responding, I guess we can reassign this issue to @balsa-asanovic. Please, let us know whether you are still interested.

Thanks!

**Comment by balsa-asanovic** [comment]
Hey @dgdavid @imobachgs ,

I wasn't sure should I tag some for pull request review, so here it is :smile: 

I've addressed this issue in PR #671

**Comment by imobachgs** [comment]
> Hey @dgdavid @imobachgs ,
> 
> I wasn't sure should I tag some for pull request review, so here it is smile

We usually keep an eye on the projects' pull request list, but feel free to assign the review to any of us just to be sure :wink:

> I've addressed this issue in PR #671

Thanks!

**Comment by balsa-asanovic** [comment]
> We usually keep an eye on the projects' pull request list, but feel free to assign the review to any of us just to be sure 😉

Great, thanks for the info!
